#2653 network-connection-timeout-problem-solved

TheVixhal opened the pull request:
#2653
Here's what has been added or modified:

New Method 'retryRequest': Added a new method 'retryRequest' to handle retry logic after a temporary network failure. This method retries the request a specified number of times ('retries' parameter) if it encounters a 'java.net.ConnectException.'

Integration of 'retryRequest' Method: Integrated the 'retryRequest' method into the 'NettyClientDriver' class.

/claim #2653

TheVixhal committed:
Update NettyClientDriver.scala

#2653
Here's what has been added or modified:

New Method 'retryRequest': Added a new method 'retryRequest' to handle retry logic after a temporary network failure. This method retries the request a specified number of times ('retries' parameter) if it encounters a 'java.net.ConnectException.'

Integration of 'retryRequest' Method: Integrated the 'retryRequest' method into the 'NettyClientDriver' class.

/claim 2653

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zio/zio-http?pullRequest=2676) <br/>All committers have signed the CLA.

TheVixhal committed:
Update NettyClientDriver.scala

TheVixhal committed:
Update NettyClientDriver.scala

#2653
/claim #2653

TheVixhal committed:
Update NettyClientDriver.scala

#2653
claim/ #2653

TheVixhal committed:
Update NettyClientDriver.scala

I replaced ZIO.effectSuspend with ZIO.suspend.
I provided explicit Trace parameters where required.
I fixed the type mismatch issue by ensuring that the block inside ZIO.suspend returns a ZIO value.
I introduced the new method retryRequest and made the necessary adjustments.

TheVixhal committed:
Update NettyClientDriver.scala

TheVixhal committed:
Update WebSocketReconnectingClient.scala

TheVixhal committed:
Update WebSocketReconnectingClient.scala

TheVixhal committed:
Update WebSocketReconnectingClient.scala

TheVixhal committed:
Update WebSocketReconnectingClient.scala

TheVixhal committed:
Update WebSocketReconnectingClient.scala
